tests/Task/LazyTaskListTest.php

```diff
@@ -119,6 +119,23 @@ public function testListCanWalkThroughTask(): void
         self::assertCount(1, $task->getTags());
     }
 
+    public function testInitializedListCanApplyMapClosure(): void
+    {
+        $list = new LazyTaskList(new TaskList());
+
+        $list->add(new NullTask('foo'));
+        $list->add(new NullTask('bar'));
+
+        self::assertSame(['foo' => 'foo', 'bar' => 'bar'], $list->map(fn (TaskInterface $task) => $task->getName()));
+    }
+
+    public function testNotInitializedListCanApplyMapClosure(): void
+    {
+        $list = new LazyTaskList(new TaskList([new NullTask('foo'), new NullTask('bar')]));
+
+        self::assertSame(['foo' => 'foo', 'bar' => 'bar'], $list->map(fn (TaskInterface $task) => $task->getName()));
+    }
+
```

Comment:
Could we add some assertions on the `isInitialized()` method? 

The lazy list should not be initialised before called calling the method then initialised right after it.

Reply:
This method is not available, should I add it?

Reply:
sorry, i will rebase main :(